configure.py

```diff
@@ -7,44 +7,58 @@
 import subprocess
 import sys
 
-
 def parse_arguments():
   parser = argparse.ArgumentParser(description="Configure the project")
-  parser.add_argument("--repo-root",
-                      help="Directory containing sources",
-                      type=str,
-                      default=os.path.abspath(os.path.dirname(__file__)))
-  parser.add_argument("--llvm-path", help="Path to llvm-project sources")
-  parser.add_argument("--iree-path", help="Path to IREE (used if enabled)")
+  parser.add_argument("--use-iree", 
+                      help="Build with IREE support and IREE's LLVM (optional)",
+                      default = False)
+  parser.add_argument("--target",
+                      help="Semicolumn-separated list of targets to build with LLVM",
+                      default = "X86")
+  parser.add_argument("--lld", 
+                      help="Build with ENABLE_LLD=ON (optional)",
+                      dest="enable_lld",
+                      default = False)
   parser.add_argument("--no-ccache",
                       help="Disables ccache (if available)",
                       dest="enable_ccache",
                       action="store_false",
                       default=True)
-  parser.add_argument("--build-dir",
-                      help="Build directory",
-                      type=str,
-                      default="build")
   return parser.parse_args()
 
 
 def main(args):
+  # Sanity checks so we are all on the same page.
+  assert "IREE_LLVM_SANDBOX_SOURCE_DIR" in os.environ, \
+    "env var IREE_LLVM_SANDBOX_SOURCE_DIR must be set"
+  assert "IREE_LLVM_SANDBOX_SOURCE_DIR" in os.environ and \
+    os.getenv("IREE_LLVM_SANDBOX_SOURCE_DIR") == os.path.abspath(os.path.dirname(__file__)), \
+    f"env var IREE_LLVM_SANDBOX_SOURCE_DIR must be set to {os.path.abspath(os.path.dirname(__file__))}"
+  assert "IREE_LLVM_SANDBOX_BUILD_DIR" in os.environ, \
```

Comment:
I generally prefer better defaults (ie. Build for in source tree) that allows override but up to you.